packages/passwordless-auth-construct/src/magic-link/magic_link_storage_service.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  DeleteCommand,
+  DynamoDBDocumentClient,
+  PutCommand,
+} from '@aws-sdk/lib-dynamodb';
+import { ConditionalCheckFailedException } from '@aws-sdk/client-dynamodb';
+import { SignedMagicLink } from '../models/magic_link.js';
+import { MagicLinkStorageConfig, StorageService } from '../types.js';
+import { logger } from '../logger.js';
+
+/**
+ * A service for storing Magic Links in DynamoDB.
+ *
+ * Links are stored with the user ID (sub) as the unique identifier.
+ */
+export class MagicLinkStorageService
+  implements StorageService<SignedMagicLink>
+{
+  /**
+   * Creates a MagicLinkStorageService instance.
+   * @param client - The DynamoDB client.
+   * @param storageConfig - The config for the service.
+   */
+  constructor(
+    private client: DynamoDBDocumentClient,
+    private storageConfig: MagicLinkStorageConfig
+  ) {}
+
+  public save = async (userId: string, magicLink: SignedMagicLink) => {
+    const { iat, exp, keyId } = magicLink;
+    const { tableName } = this.storageConfig;
+    if (!tableName) {
+      throw Error('Configuration Error: DynamoDD table must be defined.');
+    }
```

Comment:
non-blocking suggestion: `tableName` can be defined and validated in the constructor to avoid duplicated code.